sherpa/ui/tests/test_ui_unit.py

```diff
@@ -122,3 +125,83 @@ def test_save_filter_data1d(tmp_path, clean_ui):
     assert d.y == pytest.approx(expected)
     assert d.staterror is None
     assert d.syserror is None
+
+
+def test_set_iter_method_type_not_string():
+    with pytest.raises(ArgumentTypeErr) as te:
+        ui.set_iter_method(23)
+
+    assert str(te.value) == "'23' must be a string"
+
+
+def test_set_iter_method_type_not_enumeration():
+    with pytest.raises(TypeError) as te:
+        ui.set_iter_method('a random string')
+
+    assert str(te.value) == "a random string is not an iterative fitting method"
+
+
+class NonIterableObject:
```

Comment:
I sometimes use `None` for that  purpose, but on the other hand `None` has a special meaning in much of sherpa, so this is safer.